shared/js/blobview.js

```diff
@@ -420,63 +350,53 @@ var BlobView = (function() {
     },
 
     // Read bytes up to and including a null terminator, but never
-    // more than size bytes.  And return as a Latin1 string
-    readNullTerminatedLatin1Text: function(size) {
-      var s = '';
-      for (var i = 0; i < size; i++) {
-        var charcode = this.view.getUint8(this.index + i);
-        if (charcode === 0) {
-          i++;
-          break;
-        }
-        s += String.fromCharCode(charcode);
+    // more than size bytes.  And return as an ASCII string
+    readNullTerminatedLatin1Text: function(size, advance_by_size = false) {
+      var bytes = new Uint8Array(this.buffer, this.viewOffset + this.index,
+                                 size);
+
+      var nil = bytes.indexOf(0);
+      if (nil !== -1) {
+        bytes = bytes.subarray(0, nil);
       }
-      this.index += i;
+
+      var s = getDecoder('latin1').decode(bytes);
+      this.index += (nil !== -1 && !advance_by_size) ? nil + 1 : size;
       return s;
     },
 
     // Read bytes up to and including a null terminator, but never
     // more than size bytes.  And return as a UTF8 string
-    readNullTerminatedUTF8Text: function(size) {
-      for (var len = 0; len < size; len++) {
-        if (this.view.getUint8(this.index + len) === 0) {
-          break;
-        }
-      }
-      var s = this.readUTF8Text(len);
-      if (len < size) {    // skip the null terminator if we found one
-        this.advance(1);
+    readNullTerminatedUTF8Text: function(size, advance_by_size = false) {
+      var bytes = new Uint8Array(this.buffer, this.viewOffset + this.index,
+                                 size);
+
+      var nil = bytes.indexOf(0);
+      if (nil !== -1) {
+        bytes = bytes.subarray(0, nil);
       }
+
+      var s = getDecoder('utf-8').decode(bytes);
+      this.index += (nil !== -1 && !advance_by_size) ? nil + 1 : size;
```

Comment:
I suppose I could, but I wanted to avoid encouraging anyone to use a function like `getNullTerminatedByteArray`.